platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -51,42 +47,21 @@ public CrlChecker() {
         setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
     }
 
-    public HandlerResult check(BaseAuthenticationToken token,
-            X509Certificate[] certs, HandlerResult handlerResult) throws ServletException {
+    /**
+     * Checks the given certs against the CRL. The CRL is configured in this class's constructor method.
+     *
+     * @param certs certificates to be checked against the CRL
+     * @return true if one of the certs passes the CRL or if CRL revocation is disabled, false if they are all revoked.
+     */
+    public boolean passesCrlCheck(X509Certificate[] certs) {
```

Comment:
The null gets handled in the passesCrl method. If the client provides no certificates (for example, if they want to access as anonymous) then it passes the CRL check so that other handlers can attempt to handle the request.